clippy_dev/src/parse.rs

```diff
@@ -27,33 +35,101 @@ pub struct RenamedLint {
     pub version: String,
 }
 
-/// Finds all lint declarations (`declare_clippy_lint!`)
-#[must_use]
-pub fn find_lint_decls() -> Vec<Lint> {
-    let mut lints = Vec::with_capacity(1000);
-    let mut contents = String::new();
-    for e in expect_action(fs::read_dir("."), ErrAction::Read, ".") {
-        let e = expect_action(e, ErrAction::Read, ".");
-        if !expect_action(e.file_type(), ErrAction::Read, ".").is_dir() {
-            continue;
+impl ParseCxImpl {
+    /// Finds all lint declarations (`declare_clippy_lint!`)
+    #[must_use]
+    pub fn find_lint_decls(&mut self) -> Vec<Lint> {
```

Comment:
Nit: It is strange that in this commit `&mut self` is used as a parameter and never used. This would not pass dogfood should the PR be partially merged up to this commit.